src/toolbar_editor.cpp

```diff
@@ -28,10 +28,13 @@ Q_DECLARE_METATYPE(QMenu*)
 Q_DECLARE_METATYPE(QToolBar*)
 Q_DECLARE_METATYPE(QAction*)
 
+const QString Toolbar_Editor::customToolbarNamePrefix = QString("custom_toolbar_");
```

Comment:
I don't like hidden naming convention.
An alternative approach could be: allowing removal by default (without the need of an extra flag), then add a function like `void makeProtected(QString toolbar_name)` (or something like that) which prevents deletion of the matching toolbar.
Then the code that calls `setTargetWindow` would call that on toolbars that needed protection (eg: by looping through the toolbars of the main window and enforcing the naming convention in there).
Off the top of my head I can think of two possibilities for `makeProtected`:
* It could search the name in the existing toolbars and add metadata in the list item (as I suggested above)
* or it could store the names in a QSet<QString> (or something similar) and then the check is performed on whether the toolbar name is in the set.